replica/database.cc

```diff
@@ -2879,3 +2880,54 @@ const timeout_config infinite_timeout_config = {
         // not really infinite, but long enough
         1h, 1h, 1h, 1h, 1h, 1h, 1h,
 };
+
+namespace replica {
+
+future<foreign_ptr<lw_shared_ptr<reconcilable_result>>> query_mutations(
```

Comment:
I was not asked to review this unfortunately, but isn't this just a code duplication? Is it really justified? If we would make this function generic enough so storage proxy can use as well it would have make sense, but otherwise this is one more place to fixup if query code changes.

Reply:
Yes and no. Yes, it duplicates some code, but it is just an `if` and dispatch to other methods + some boilerplate, because C++.

That said, we can make storage proxy call this code, to de-duplicate even that. Seems to me like too much effort for little gain, but I don't oppose it.

Reply:
Functions indeed are not very long, but if `read_command` is extended with new parameters or `query_mutations` is extended to return more values (both things that happened in the past) those are two additional places that needs to be changed. I do see a value in calling them from the sp.